src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,365 @@
+    children.append(
+        create_dataset(
+            name="event_time_zero",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_time_offset
+    children.append(
+        create_dataset(
+            name="event_time_offset",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_index
+    children.append(
+        create_dataset(
+            name="event_index",
+            dtype="int",
+            initial_values=np.asarray([]),
+        )
+    )
+    # event_id
+    if "monitor" in group["name"]:
+        ...  # Monitor doesn't have ``event_id``
+    else:
+        children.append(
+            create_dataset(
+                name="event_id",
+                dtype="int",
+                initial_values=np.asarray([]),
             )
+        )
+
+
+def _merge_ev44(group: DataModuleParent, data_piece: Mapping) -> None:
+    """Merges new values from a message into the data group.
+
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
+    data_piece:
+        New message containing deserialized ev44.
+
+    Side Effects
+    ------------
+    Each dataset in the children of ``group`` will have new values appended.
+    Most of array-like values are simply appended
+    but the ``event_index``(``reference_time_index`` from ev44) is increased
+    by the number of previous values of ``event_time_offset``
+    to find the global ``event_index``.
+    ``data_piece`` only has local ``event_index`` which always starts with 0.
+
+    """
+    # event_time_zero - reference_time
+    event_time_zero_dataset = find_nexus_structure(group, ("event_time_zero",))
+    original_event_time_zero = _get_array_values(event_time_zero_dataset)
+    new_event_time_zero = data_piece["reference_time"]
+    _set_values(
+        event_time_zero_dataset,
+        np.concatenate((original_event_time_zero, new_event_time_zero)),
+    )
+    # event_time_offset - time_of_flight
+    event_time_offset_dataset = find_nexus_structure(group, ("event_time_offset",))
+    original_event_time_offset = _get_array_values(event_time_offset_dataset)
+    new_event_time_offset = data_piece["time_of_flight"]
+    _set_values(
+        event_time_offset_dataset,
+        np.concatenate((original_event_time_offset, new_event_time_offset)),
+    )
+    # event_index - reference_time_index
+    event_index_dataset = find_nexus_structure(group, ("event_index",))
+    original_event_index = _get_array_values(event_index_dataset)
+    event_index_offset = len(original_event_time_offset)
+    new_event_index = data_piece["reference_time_index"] + event_index_offset
+    _set_values(
+        event_index_dataset, np.concatenate((original_event_index, new_event_index))
+    )
+    # event_id - pixel_id
+    if (
+        "pixel_id" in data_piece and data_piece["pixel_id"] is not None
```

Comment:
(minor) `data_piece.get("pixel_id")` is shorthand for the check in the if statement